### ISSUE TITLE
Should not report SC2090 when reporting SC2086 on unquoted variable in 'eval cmd $var'.

### ISSUE THREAD

**Issue by deryni** [koalaman/shellcheck#63]
I realize this is a minor point because fixing the SC2086 violation will cause both errors to go away but SC2090 is a false positive in this case as the embedded quotes will work correctly.

Shellcheck will report both SC2086 and SC2090 for cases 1, 2 and 3 below.

``` bash
#!/bin/bash

c() {
    echo "Arg count: $#"
    for arg; do
        echo "\$arg: $arg"
    done
}

args="--foo 'bar baz' 't*'"

echo 1
c "$args"
echo 2
c $args
echo 3
eval c $args
echo 4
eval c "$args"
```
